fix(deps): bump zlob to 1.3.2 for zig 0.16 compatibility

### Description

Local `cargo run` / `just build` fails on dev boxes with zig 0.16.0 because `Cargo.lock` pins `zlob 1.3.0`, whose `build.zig` uses pre-0.16 APIs (`linkLibC()` method call on `Build.Step.Compile` — removed in 0.16).

```
build.zig:399:25: error: no field or member function named 'linkLibC' in 'Build.Step.Compile'
            compare_libc.linkLibC(); // Link against libc for glob()
```

`zlob 1.3.2` is the first version that targets zig 0.16's API. `fff-search` (via `ea1f980`) declares `zlob = "1.3.0"` which under semver allows 1.3.2, so only `Cargo.lock` needs updating.

This is the **local-side counterpart** to the CI zig toolchain issue — CI runner has 0.15.2 (separate infra task), but every dev box with homebrew zig is now on 0.16.

### Component

ci (CI/CD, build, release)

### Steps to reproduce

1. `brew install zig` (gets 0.16.0)
2. `cargo build` in repo root
3. `zlob v1.3.0` fails to compile

### Logs / Error output

See above.

### Version

rara HEAD (2026-04-21)

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -288,6 +288,13 @@ fff-search = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f980
 fff-grep = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f9802d7879345f02bef72bcadecd7252b8add" }
 fff-query-parser = { git = "https://github.com/dmtrKovalenko/fff.nvim", rev = "ea1f9802d7879345f02bef72bcadecd7252b8add", default-features = false }
 
+# zlob is a transitive dep of fff-search (feature = "zlob"); declaring it here
+# pins the version across the workspace because `Cargo.lock` is gitignored.
+# 1.3.2 is the first zlob release targeting zig 0.16's `Build.Step.Compile`
+# API (the `linkLibC()` method call was removed upstream); 1.3.0 only compiles
+# on zig 0.15. Bump this in lockstep with the CI runner's zig toolchain.
+zlob = "=1.3.2"
+
 rara-tool-macro = { path = "crates/common/tool-macro" }
 yunara-store = { path = "crates/common/yunara-store" }
 

--- a/crates/app/Cargo.toml
+++ b/crates/app/Cargo.toml
@@ -57,6 +57,9 @@ rara-soul.workspace = true
 rara-tool-macro.workspace = true
 fff-search = { workspace = true }
 fff-query-parser = { workspace = true }
+# Direct dep (not imported) to activate the workspace-level `=1.3.2` pin on
+# this transitive; see the note in the workspace `[dependencies]`.
+zlob = { workspace = true }
 regex.workspace = true
 reqwest = { workspace = true }
 schemars.workspace = true

